packages/angular/src/table/table-model.class.ts

```diff
@@ -996,8 +984,135 @@ export class AITableModel implements PaginationModel {
     } else {
       // if moving to right
       const block = list.slice(blockStart, blockEnd + 1);
-      list.splice.apply(list, [index, 0].concat(block));
+      list.splice.apply(list, [index + 1, 0].concat(block));
       list.splice(blockStart, blockEnd - blockStart + 1);
     }
   }
+
+  protected tabularToNested(
+    header: TableHeaderItem[][],
+    data: TableItem[][],
+    headerRow: TableHeaderItem[] = [],
+    availableHeaderItems: TableHeaderItem[][] = [],
+    leafIndex = { index: 0 }, // So we can pass by reference
```

Comment:
```suggestion
    // This allows us to walk the leaves as if they were in a list from left to right.
    // We need to pass by reference so that we can update this value from within the recursion.
    leafIndex = { index: 0 },
```